lib/guest_installation_and_configuration_base.pm

```diff
@@ -779,119 +903,338 @@ sub config_guest_rng {
+        $self->config_guest_network_bridge($self->{guest_network_device}, $_ipaddr . '/24', $self->{guest_domain_name});
+        $self->config_guest_network_bridge_policy($self->{guest_network_device});
+        $self->{guest_network_selection_options} = "--network=bridge=$self->{guest_network_device},mac=$self->{guest_macaddr}";
+    }
+    save_screenshot;
+    record_info("Guest network configuration done", script_output("ip addr show;ip route show all;virsh net-list --all;(for i in \`virsh net-list --all --name\`;do virsh net-dumpxml \$i;done);ps axu | grep dnsmasq", type_command => 1, proceed_on_failure => 1));
+    return $self;
+}
+
+=head2 config_guest_macaddr
+
+  config_guest_macaddr($self)
+
+Generate nearly random mac address.
+
+=cut
+
+sub config_guest_macaddr {
+    my $self = shift;
+
+    $self->reveal_myself;
+    my $_guest_macaddr_lower_half = join ':', map { unpack "H*", chr(rand(256)) } 1 .. 3;
+    if ($self->{guest_network_type} eq 'bridge' and $self->{guest_network_mode} eq 'host') {
+        $self->{guest_macaddr} = 'd4:c9:ef:' . $_guest_macaddr_lower_half;
+    }
+    else {
+        $self->{guest_macaddr} = '52:54:00:' . $_guest_macaddr_lower_half;
+    }
+    return $self;
+}
+
+=head2 config_guest_network_vnet
+
+  config_guest_network_vent($self[, _device => 'device', _ipaddr => 'ip', 
+  _netmask => 'mask', _startaddr => 'start', _endaddr => 'end'])
+
+Create virtual network to be used with guest based on [guest_network_device] and
+passed in arguments, $_ipaddr, $_netmask, $_startaddr and $_endaddr. Skip creating
+already existing and active virtual network.
+
+=cut
+
+sub config_guest_network_vnet {
+    my ($self, %args) = @_;
+    $args{_device} //= '';
+    $args{_ipaddr} //= '';
+    $args{_netmask} //= '';
+    $args{_startaddr} //= '';
+    $args{_endaddr} //= '';
+
+    $self->reveal_myself;
+    if (!$args{_device} or !$args{_ipaddr} or !$args{_netmask} or !$args{_startaddr} or !$args{_endaddr}) {
+        croak("Network device, ip address, network mask, start and end address must be given to create virtual network for guest $self->{guest_name}");
+    }
+    unless (script_run("virsh net-list | grep \"$args{_device} .*active\"") == 0) {
+        script_run("virsh net-destroy $args{_device}");
+        if ($args{_device} ne 'default' or script_run("virsh net-list --all --name | grep -i default") != 0) {
+            script_run("virsh net-undefine $args{_device}");
+            my $_forward_mode = ($self->{guest_network_mode} eq 'default') ? 'nat' : $self->{guest_network_mode};
+            type_string("cat > /var/lib/libvirt/images/$args{_device}.xml <<EOF
+<network>
+  <name>$args{_device}</name>
+  <bridge name=\"$args{_device}\"/>
+EOF
+");
+            if ($_forward_mode eq 'nat') {
+                type_string("cat >> /var/lib/libvirt/images/$args{_device}.xml <<EOF
+  <forward mode=\"$_forward_mode\">
+    <nat>
+      <port start=\"20232\" end=\"65535\"/>
```

Comment:
This seems different from recommended nat vnet, https://documentation.suse.com/sles/15-SP5/single-html/SLES-virtualization/#ex-libvirt-net-nat